src/core/Akka/Actor/Props.cs

```diff
@@ -285,37 +155,51 @@ public Props(Deploy deploy, Type type, IEnumerable<object> args)
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="Props" /> class.
+        ///     Initializes a new instance of the <see cref="Props" /> class.
         /// </summary>
         /// <param name="deploy">The configuration used to deploy the actor.</param>
         /// <param name="type">The type of the actor to create.</param>
         /// <param name="args">The arguments needed to create the actor.</param>
-        /// <exception cref="ArgumentException">This exception is thrown if <paramref name="type"/> is an unknown actor producer.</exception>
-        public Props(Deploy deploy, Type type, params object[] args)
+        /// <exception cref="ArgumentException">This exception is thrown if <paramref name="type" /> is an unknown actor producer.</exception>
+        public Props(Deploy deploy, Type type, params object[] args) 
+            : this(CreateProducer(type, args), deploy, args) // have to preserve the "CreateProducer" call here to preserve backwards compat with Akka.DI.Core
+        {
+
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="Props" /> class using a specified <see cref="IIndirectActorProducer"/>.
+        /// </summary>
+        /// <remarks>
+        ///     This API is meant for advanced use cases, such as Akka.DependencyInjection.
+        /// </remarks>
+        /// <param name="producer">The type of <see cref="IIndirectActorProducer"/> that will be used to instantiate <see cref="Type"/></param>
+        /// <param name="deploy">The configuration used to deploy the actor.</param>
+        /// <param name="args">The arguments needed to create the actor.</param>
+        internal Props(IIndirectActorProducer producer, Deploy deploy, params object[] args)
         {
             Deploy = deploy;
-            inputType = type;
-            Arguments = args ?? noArgs;
-            producer = CreateProducer(inputType, Arguments);
+            _inputType = producer.ActorType;
+            Arguments = args ?? NoArgs;
+           _producer = producer;
```

Comment:
`IIndirectActorProducer` is now passed in directly, rather than instantiated via reflection magic.